### PR TITLE
aoscx: Fix fan RPM speeds included in environment output

### LINKED ISSUE #2864: aoscx: Fan RPM information included, causing constant diffs
There have been various issues mentioning this like #2612 [pull-request#2565](https://github.com/ytti/oxidized/pull/2565) but none merged and the various fixes are not sufficient to stop all of them.

Example from a 6300M
```
! Fan tray information
! ------------------------------------------------------------------------------
! Name  Description                           Status        Serial Number  Fans
! ------------------------------------------------------------------------------
! 1/1   JL669A Aruba X751 FB Fan Tray         ready         CN04KN9P4S     2  
! 1/2   JL669A Aruba X751 FB Fan Tray         ready         CN04KN9P3R     2  
! 2/1   JL669A Aruba X751 FB Fan Tray         ready         CN04KN9PMJ     2  
! 2/2   JL669A Aruba X751 FB Fan Tray         ready         CN04KN9LR3     2  
! 
! Fan information
! ------------------------------------------------------------------------------
! Location-      Product  Serial Number  Speed   Direction      Status  RPM
! Mbr/Slot/Fan   Name
! ------------------------------------------------------------------------------
! Tray-1/1/1     N/A      N/A            slow    front-to-back  ok      3994   
! Tray-1/1/2     N/A      N/A            slow    front-to-back  ok      3983   
! Tray-1/2/1     N/A      N/A            slow    front-to-back  ok      3971   
! Tray-1/2/2     N/A      N/A            slow    front-to-back  ok      4039   
! Tray-2/1/1     N/A      N/A            slow    front-to-back  ok      3963   
! Tray-2/1/2     N/A      N/A            slow    front-to-back  ok      3949   
! Tray-2/2/1     N/A      N/A            slow    front-to-back  ok      3942   
! Tray-2/2/2     N/A      N/A            slow    front-to-back  ok      3959   
! System-3/1/1   N/A      N/A            slow    left-to-back   ok      3994   
! System-3/1/2   N/A      N/A            slow    left-to-back   ok      13374  
! System-3/1/3   N/A      N/A            slow    left-to-back   ok      3998   
! 
```

Pull request incoming enhancing the other requests

### PR DESCRIPTION
**Pull request by danpoltawski**
## Pre-Request Checklist

- [x] Passes rubocop code analysis (try `rubocop --auto-correct`)
- [x] Tests added or adapted (try `rake test`)
- [ ] Changes are reflected in the documentation
- [x] User-visible changes appended to [CHANGELOG.md](/CHANGELOG.md)

## Description
Removes fan speed from Aruba CX output. Build on the work by @vppencilsharpener and @steveneppler

Before:
```
! Fan information
! ------------------------------------------------------------------------------
! Location-      Product  Serial Number  Speed   Direction      Status  RPM
! Mbr/Slot/Fan   Name
! ------------------------------------------------------------------------------
! Tray-1/1/1     N/A      N/A            slow    front-to-back  ok      3961   
! Tray-1/1/2     N/A      N/A            slow    front-to-back  ok      3987   
! Tray-1/2/1     N/A      N/A            slow    front-to-back  ok      4016   
! Tray-1/2/2     N/A      N/A            slow    front-to-back  ok      3998   
! Tray-2/1/1     N/A      N/A            slow    front-to-back  ok      3973   
! Tray-2/1/2     N/A      N/A            slow    front-to-back  ok      3990   
! Tray-2/2/1     N/A      N/A            slow    front-to-back  ok      3996   
! Tray-2/2/2     N/A      N/A            slow    front-to-back  ok      4008   
! System-3/1/1   N/A      N/A            slow    left-to-back   ok      4024   
! System-3/1/2   N/A      N/A            slow    left-to-back   ok      13374  
! System-3/1/3   N/A      N/A            slow    left-to-back   ok      3990   
```

After:
```
! Fan information
! ------------------------------------------------------------------------------
! Location-      Product  Serial Number  Speed   Direction      Status  RPM
! Mbr/Slot/Fan   Name
! ------------------------------------------------------------------------------
! Tray-1/1/1     N/A      N/A            <speed>    front-to-back  ok      <rpm>   
! Tray-1/1/2     N/A      N/A            <speed>    front-to-back  ok      <rpm>   
! Tray-1/2/1     N/A      N/A            <speed>    front-to-back  ok      <rpm>   
! Tray-1/2/2     N/A      N/A            <speed>    front-to-back  ok      <rpm>   
! Tray-2/1/1     N/A      N/A            <speed>    front-to-back  ok      <rpm>   
! Tray-2/1/2     N/A      N/A            <speed>    front-to-back  ok      <rpm>   
! Tray-2/2/1     N/A      N/A            <speed>    front-to-back  ok      <rpm>   
! Tray-2/2/2     N/A      N/A            <speed>    front-to-back  ok      <rpm>   
! System-3/1/1   N/A      N/A            <speed>    left-to-back   ok      <rpm>   
! System-3/1/2   N/A      N/A            <speed>    left-to-back   ok      <rpm>  
! System-3/1/3   N/A      N/A            <speed>    left-to-back   ok      <rpm>   
! 
```

Closes #2864

### CODE REVIEW

File: lib/oxidized/model/aoscx.rb [hunk 1]
```diff
@@ -58,6 +58,7 @@ class Aoscx < Oxidized::Model
     cfg.gsub! /^(\d\/\d\/\d.*\s+)\d+\s+$/, '\\1<hidden>'
     cfg.gsub! /^(\d+\/\S+\s+\S+\s+)\d+\.\d+\s+C(.*)/, '\\1<hidden>\\2'
     cfg.gsub! /^(LC.*\s+)\d+\.\d+\s+(C.*)$/, '\\1 <hidden> \\2'
+    cfg.gsub! /^((System-|Tray-)?\d+\/\S+\s+\S+\s+\S+\s+)(\S+)(\s+\S+\s+\S+\s+)\d+/, '\\1<speed>\\4<rpm>'
```
Comment (rmounce) [inline lib/oxidized/model/aoscx.rb:61]:
```suggestion
    cfg.gsub! /^(\w+-\d+\/\S+\s+\S+\s+\S+\s+)(\S+)(\s+\S+\s+\S+\s+)\d+/, '\\1<speed>\\3<rpm>'
```

**Review by rmounce — CHANGES_REQUESTED** [review]
Thanks Dan. On our 8360-24XF2C this proposed fix doesn't match all fans, and erroneously matches static power supply information. I've tested a refined regex.

Original
```
! Fan information
! ------------------------------------------------------------------------------
! Location-      Product  Serial Number  Speed   Direction      Status  RPM
! Mbr/Slot/Fan   Name
! ------------------------------------------------------------------------------
! PSU-1/1/1      N/A      N/A            slow    back-to-front  ok      6224   
! PSU-1/2/1      N/A      N/A            slow    back-to-front  ok      6240   
! Tray-1/1/1     N/A      N/A            slow    back-to-front  ok      5821   
! Tray-1/1/2     N/A      N/A            slow    back-to-front  ok      5201   
! Tray-1/2/1     N/A      N/A            slow    back-to-front  ok      5803   
! Tray-1/2/2     N/A      N/A            slow    back-to-front  ok      5093   
! Tray-1/3/1     N/A      N/A            slow    back-to-front  ok      5719   
! Tray-1/3/2     N/A      N/A            slow    back-to-front  ok      5110   
! 
! show environment power-supply
! ------------------------------------------------------------------------------
!          Product  Serial            PSU            Input   Voltage    Wattage
! Mbr/PSU  Number   Number            Status         Type    Range      Maximum
! ------------------------------------------------------------------------------
! 1/1      JL712A   CNREDACTED        OK             AC      100V-240V  550
! 1/2      JL712A   CNREDACTED        OK             AC      100V-240V  550
```

`cfg.gsub! /^((System-|Tray-)?\d+\/\S+\s+\S+\s+\S+\s+)(\S+)(\s+\S+\s+\S+\s+)\d+/, '\\1<speed>\\4<rpm>'`
```
! Fan information
! ------------------------------------------------------------------------------
! Location-      Product  Serial Number  Speed   Direction      Status  RPM
! Mbr/Slot/Fan   Name
! ------------------------------------------------------------------------------
! PSU-1/1/1      N/A      N/A            slow    back-to-front  ok      6224   
! PSU-1/2/1      N/A      N/A            slow    back-to-front  ok      6256   
! Tray-1/1/1     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! Tray-1/1/2     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! Tray-1/2/1     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! Tray-1/2/2     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! Tray-1/3/1     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! Tray-1/3/2     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! 
! show environment power-supply
! ------------------------------------------------------------------------------
!          Product  Serial            PSU            Input   Voltage    Wattage
! Mbr/PSU  Number   Number            Status         Type    Range      Maximum
! ------------------------------------------------------------------------------
! 1/1      JL712A   CNREDACTED        <speed>             AC      100V-240V  <rpm>
! 1/2      JL712A   CNREDACTED        <speed>             AC      100V-240V  <rpm>
```

`cfg.gsub! /^(\w+-\d+\/\S+\s+\S+\s+\S+\s+)(\S+)(\s+\S+\s+\S+\s+)\d+/, '\\1<speed>\\3<rpm>'`
```
! Fan information
! ------------------------------------------------------------------------------
! Location-      Product  Serial Number  Speed   Direction      Status  RPM
! Mbr/Slot/Fan   Name
! ------------------------------------------------------------------------------
! PSU-1/1/1      N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! PSU-1/2/1      N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! Tray-1/1/1     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! Tray-1/1/2     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! Tray-1/2/1     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! Tray-1/2/2     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! Tray-1/3/1     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! Tray-1/3/2     N/A      N/A            <speed>    back-to-front  ok      <rpm>   
! 
! show environment power-supply
! ------------------------------------------------------------------------------
!          Product  Serial            PSU            Input   Voltage    Wattage
! Mbr/PSU  Number   Number            Status         Type    Range      Maximum
! ------------------------------------------------------------------------------
! 1/1      JL712A   CNREDACTED        OK             AC      100V-240V  550
! 1/2      JL712A   CNREDACTED        OK             AC      100V-240V  550
```

**Comment by danpoltawski** [comment]
Ah no, but this is not matching this case:
https://github.com/ytti/oxidized/issues/2864#issuecomment-1632247001

I feel like a change of approach might be needed to match on `slow|fast` (and other 'speed' values, but I'm not sure what the entire list would be)

**Comment by danpoltawski** [comment]
> I feel like a change of approach might be needed to match on `slow|fast` (and other 'speed' values, but I'm not sure what the entire list would be)

https://www.arubanetworks.com/techdocs/AOS-CX/10.07/HTML/5200-7847/Content/SysHW_cmds/sho-env-fan-10.htm

**Comment by danpoltawski** [comment]
@rmounce I'd be interested if this patch works for your devices too, I feel its more robust but I'm relying on the suggested values in the docs for the different fan speeds

**Comment by rmounce** [comment]
@danpoltawski
Works for me. The list of fan speeds from that doc looks comprehensive, seems like a reasonable approach.

**Comment by eoli3n** [comment]
When is the new release planned to include that fix ?

**Pull request merged**